src/Tokenizer/TokensAnalyzer.php

```diff
@@ -535,8 +535,8 @@ private function findClassyElements(Tokens $tokens, $start)
                 continue;
             }
 
-            if (!$inClass) {
-                $inClass = $token->isClassy();
+            if (false === $inClass) {
+                $inClass = $token->isClassy() ? $index : false;
```

Comment:
concern (a little out of the blue):
does this method supports anonymous classes?

Reply:
yes it does, anonymous classes use the same `T_CLASS` token as regular classes;
```bash
$ php test.php 

    0 | T_OPEN_TAG                              | <?php

    1 | T_WHITESPACE                            | 

    2 | T_VARIABLE                              | $a
    3 | T_WHITESPACE                            |  
    4 |                                         | =
    5 | T_WHITESPACE                            |  
    6 | T_NEW                                   | new
    7 | T_WHITESPACE                            |  
    8 | T_CLASS                                 | class
    9 | T_WHITESPACE                            |  
   10 |                                         | {
   11 | T_PUBLIC                                | public
```

Reply:
it's not question in terms of `isClassy` implementation details, but fixer implementation and coverage by tests.

Reply:
this is the analyzer code, so I don't understand, it is covered by the analyzer tests

Reply:
> fixer implementation and coverage by tests.

tests added for `anonymous classes` for the fixer implementation